mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/TaskExecutorState.java

```diff
@@ -100,10 +100,9 @@ boolean onRegistration(TaskExecutorRegistration registration) {
             this.registration = registration;
             this.gateway =
                 rpcService.connect(registration.getTaskExecutorAddress(), TaskExecutorGateway.class)
-                    .whenComplete((gateway, throwable) -> {
-                        if (throwable != null) {
-                            log.error("Failed to connect to the gateway", throwable);
-                        }
+                    .exceptionally((throwable) -> {
+                        log.error("Failed to connect to the gateway onRegistration, reset to unregistered.", throwable);
```

Comment:
I am under the impression that an exception is supposed to be thrown when onRegistration encounters a failure. In fact, onTaskExecutorRegistration incorporates a try/catch block around it. If we decide not to throw any exceptions now and state.onRegistration(registration); gives a true value, could this result in a semantic shift? Are all possible scenarios addressed correctly in this situation?

Reply:
the try-catch on RC actor won't be available to catch this exception at all as it wasn't part of the returned value.

Reply:
the exception will happen to another thread later time and RC actor won't see it either.